DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.4_Recursive_Tree_Method/1.2.2.4_Recursive_Tree_Method.md

```diff
@@ -0,0 +1,87 @@
+# Recursion Tree Method
+
+While the substitution method is a popular technique to find time complexity of recursive algorithms, recursion tree method is a useful technique to solve divide and conquer based recursive algorithms.
+
+## Generic Process
+
+We break down a recurrence T(n) into subproblems until we reach T(1). Here, each node represents the cost of a subproblem. We sum up the cost of each node to find out the total cost of the algorithm.
+
+The following steps are involved:
+
+![](Images/generic_tree.png)
+
+1. The Recursion Tree for a given recurrence relation is drawn. At every level, the parent problem is divided into multiple subproblems, until cost of the subproblem becomes 1.
+
+2. We find the total number of leaf nodes and the total cost of the last level.
+
+3. We find a relation to determine the total cost of the kth level. Total cost of level 1 to k-1 is calculated.
+
+4. Total Cost of algorithm = Total cost of leaf nodes + Total cost of internal nodes.
```

Comment:
Highlight this statement
eg->  `Total cost of leaf nodes + Total cost of internal nodes.`

Reply:
@harshsri28 Updated and highlighted steps under generic process.